dio/lib/src/dio_mixin.dart

```diff
@@ -556,18 +556,18 @@ abstract class DioMixin implements Dio {
       );
       final statusOk = reqOpt.validateStatus(responseBody.statusCode);
       if (statusOk || reqOpt.receiveDataWhenStatusError == true) {
-        final forceConvert = !(T == dynamic || T == String) &&
-            !(reqOpt.responseType == ResponseType.bytes ||
-                reqOpt.responseType == ResponseType.stream);
-        String? contentType;
-        if (forceConvert) {
-          contentType = headers.value(Headers.contentTypeHeader);
-          headers.set(Headers.contentTypeHeader, Headers.jsonContentType);
+        // Imply as JSON if the given generic type matches the condition.
+        final implicitConvert = T != dynamic &&
```

Comment:
`forceConvert` should be disabled when receiving an error status code

```suggestion
        final implicitConvert = statusOk && T != dynamic &&
```